etc/scripts/get-stack.sh

```diff
@@ -748,11 +748,70 @@ on_path() {
 # Check whether ~/.local/bin is on the PATH, and print a warning if not.
 check_home_local_bin_on_path() {
   if ! on_path "$HOME_LOCAL_BIN" ; then
-    #TODO: offer to add it for the user (pull requests welcome!)
     info "WARNING: '$HOME_LOCAL_BIN' is not on your PATH."
     info "    Stack will place the binaries it builds in '$HOME_LOCAL_BIN' so"
     info "    for best results, please add it to the beginning of PATH in your profile."
     info ""
+    # detect which profile file to use, then print a message about updating it
+    if [ -n "$BASH_VERSION" ]; then
+      if [ -f "$HOME/.bash_profile" ]; then
+        profile_file="$HOME/.bash_profile"
+      else
+        profile_file="$HOME/.bashrc"
+      fi
+    elif [ -n "$ZSH_VERSION" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -n "$FISH_VERSION" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -n "$PROFILE" ]; then
+      profile_file="$PROFILE"
+    elif [ -f "$HOME/.bash_profile" ]; then
+      profile_file="$HOME/.bash_profile"
+    elif [ -f "$HOME/.bashrc" ]; then
+      profile_file="$HOME/.bashrc"
+    elif [ -f "$HOME/.zshrc" ]; then
+      profile_file="$HOME/.zshrc"
+    elif [ -f "$HOME/.config/fish/config.fish" ]; then
+      profile_file="$HOME/.config/fish/config.fish"
+    elif [ -f "$HOME/.profile" ]; then
+      profile_file="$HOME/.profile"
+    else
+      info "    (profile not found; please add it to your PATH manually)"
+    fi
+
+    # if we found a profile file, print a message about updating it
+    if [ -n "$profile_file" ]; then
+      info "    You can do this by running the following command:"
+      info "    echo 'export PATH=\"$HOME_LOCAL_BIN:\$PATH\"' >> \"$profile_file\""
+      info "    (You may need to restart your shell for this to take effect.)"
+      info ""
+    fi
+    # if we found a profile file and we're not in quiet mode, prompt user to update it
+    if [ -z "$QUIET" ]; then
```

Comment:
Despite what line 789 says, this part seems to me to be outside the test for the existence of contents of `profile_file` at line 783, given the `fi` at line 788?

This shell script is not expected to be used on Windows except, possibily, as part of CI (see line 275) but I assume on Windows there is no 'profile' file and so it is important that the prompt is within the scope of the test.

Reply:
sorry about that. careless oversight. fixed.